src/compiler/program.ts

```diff
@@ -1877,6 +1930,23 @@ namespace ts {
             return file;
         }
 
+        function getProjectReferenceRedirect(fileName: string): string | undefined {
+            const path = toPath(fileName);
+            // If this file is produced by a referenced project, we need to rewrite it to
+            // look in the output folder of the referenced project rather than the input
+            const normalized = getNormalizedAbsolutePath(fileName, path);
+            let result: string | undefined;
+            projectReferenceRedirects.forEach((v, k) => {
+                if (result !== undefined) {
+                    return undefined;
+                }
+                if (normalized.indexOf(k) === 0) {
+                    result = changeExtension(fileName.replace(k, v), ".d.ts");

```

Comment:
we do have a function to get the declaration file path, can we use that instead?

Reply:
`getDeclarationEmitOutputFilePath` is what i was referring to.

Reply:
That function accepts a `SourceFile`, which we definitely don't have

Reply:
But it would be easy to modify since only sourceFile.fileName is used in there anyways.